cmd/necogcp/cmd/createrunner.go

```diff
@@ -0,0 +1,199 @@
+	labels         []string
+)
+
+var runnerStartUpScriptTmpl = template.Must(template.New("").Parse(`#!/bin/sh
+
+# Wait for auto-refresh to be done
+snap watch --last=auto-refresh
+
+useradd -m runner
+usermod -aG google-sudoers runner
+RUNNER_HOME=/home/runner
+/snap/bin/gcloud secrets versions access latest --secret="ghatoken_{{ .RunnerName }}" > $RUNNER_HOME/github_token
+cat << EOF >job-cancelled
+#!/bin/sh
+EOF
+chmod 755 job-cancelled
+cp job-cancelled /usr/local/bin/job-started
+cp job-cancelled /usr/local/bin/job-failure
+cp job-cancelled /usr/local/bin/job-success
+mv job-cancelled /usr/local/bin/job-cancelled
+
+NUM_DEVICES=$(ls /dev/nvme0n*|wc -l)
+mdadm --create /dev/md0 -l stripe --raid-devices=${NUM_DEVICES} /dev/nvme0n*
+mkfs -t ext4 -F /dev/md0
+mkdir -p /var/scratch
+mount -t ext4 /dev/md0 /var/scratch
+chmod 1777 /var/scratch
+
+latest=$(curl -sSf https://api.github.com/repos/actions/runner/releases/latest | jq -r '.tag_name' | sed -e 's/v//')
+curl -L -o $RUNNER_HOME/runner.tar.gz https://github.com/actions/runner/releases/download/v"$latest"/actions-runner-linux-x64-"$latest".tar.gz
+tar xzf $RUNNER_HOME/runner.tar.gz -C $RUNNER_HOME
+$RUNNER_HOME/bin/installdependencies.sh
+
+cat << 'EOT' >$RUNNER_HOME/run-actions.sh
+#!/bin/sh
+REPO_URL=https://github.com/{{ .RunnerRepo }}
+
+export NECO_DIR=${GOPATH}/src/github.com/cybozu-go/neco
+export NECO_APPS_DIR=${GOPATH}/src/github.com/cybozu-private/neco-apps
+
+$HOME/config.sh \
+	--unattended \
+	--replace \
+	--name $(hostname) \
+	--labels {{ .LabelOption }} \
+	--url $REPO_URL \
+	--token $(cat $HOME/github_token) \
+	--work $HOME \
+	--ephemeral
+
+code=-1
+while  [ "$code" -ne 1 ] && [ "$code" -ne 0 ]
+do
+	$HOME/bin/Runner.Listener run --startuptype service
+	code=$?
+done
+
+EOT
+
+chmod 755 $RUNNER_HOME/run-actions.sh
+sudo -i -u runner $RUNNER_HOME/run-actions.sh
+`))
+
+var createRunnerCmd = &cobra.Command{
+	Use:   "create-runner",
+	Short: "Launch runner instance which runs self-hosted-runner",
+	Long: `Launch runner instance which runs self-hosted-runner with vmx-enabled image.
+
+If runner instance already exists in the project, new runner is not created.`,
+	Args: cobra.ExactArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		splitRunnerRepo := strings.Split(runnerRepo, "/")
+		owner := splitRunnerRepo[0]
+		repository := splitRunnerRepo[1]
+		secretName := fmt.Sprintf("%s_%s", secretPrefix, runnerName)
+		lableOption := strings.Join(append(labels, defaultRunnerLabel), ",")
+
+		buf := new(bytes.Buffer)
+		err := runnerStartUpScriptTmpl.Execute(buf, struct {
+			RunnerName  string
+			RunnerRepo  string
+			LabelOption string
+		}{
+			RunnerName:  runnerName,
+			RunnerRepo:  runnerRepo,
+			LabelOption: lableOption,
+		})
+		if err != nil {
+			log.ErrorExit(err)
+		}
+		startupScript := buf.String()
+
+		well.Go(func(ctx context.Context) error {
+			secretClient, err := gcp.NewSecretClient(ctx, runnerProjectId)
+			if err != nil {
+				log.ErrorExit(err)
```

Comment:
We need not call `log.ErrorExit()` in `well.Go`. It is enough to return an error.

If a function given as an argument of `well.Go` returns an error, the following logic will output an error log and exit.

```
		err = well.Wait()
		if err != nil {
			log.ErrorExit(err)
		}
```